File: src/main/java/com/ennoblesoft/demo/controller/StudentController.java
package com.ennoblesoft.demo.controller;

import com.ennoblesoft.demo.entity.Student;
import com.ennoblesoft.demo.exception.CustomException;
import com.ennoblesoft.demo.service.StudentService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

import static org.springframework.http.HttpStatus.OK;

@RestController
@RequestMapping("/api/v1")
public class StudentController {
    @Autowired
    StudentService studentService;

    @GetMapping("/student")
    public List<Student> get(@RequestParam("name") String name) throws CustomException {
        List<Student> studentList = studentService.get(name);
        if (studentList.isEmpty())
            throw new CustomException("No Student Found");
        else
            return studentList;
    }

    @PostMapping("/student")
    @ResponseStatus(value = OK)
    public Student create(@RequestBody Student s) {
        return studentService.create(s);
    }
}

File: src/main/kotlin/com/ennoblesoft/springdemo/controller/StudentController.kt
package com.ennoblesoft.springdemo.controller

import com.ennoblesoft.springdemo.entity.Student
import com.ennoblesoft.springdemo.exception.CustomException
import com.ennoblesoft.springdemo.service.StudentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus.OK
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1")
class StudentController {
    @Autowired
    internal var studentService: StudentService? = null

    @GetMapping("/student")
    @Throws(CustomException::class)
    operator fun get(@RequestParam("name") name: String): List<Student>? {
        val studentList = studentService?.get(name)
        return if (studentList != null && studentList.isEmpty())
            throw CustomException("No Student Found")
        else
            studentList
    }

    @PostMapping("/student")
    @ResponseStatus(value = OK)
    fun create(@RequestBody s: Student): Student {
        return studentService!!.create(s)
    }
}
